src/libraries/System.Runtime.Numerics/tests/BigInteger/BigIntegerToStringTests.cs

```diff
@@ -439,6 +439,25 @@ public static void CustomFormatPerMille()
             RunCustomFormatToStringTests(s_random, "#\u2030000000", CultureInfo.CurrentCulture.NumberFormat.NegativeSign, 6, PerMilleSymbolFormatter);
         }
 
+        public static IEnumerable<object[]> RunFormatScientificNotationToBigIntegerAndViceVersaData()
```

Comment:
There should also be a test covering 2147483639, 2147483647, and a couple arbitrary strings bigger than that.

The latter two should end up returning false for TryParse and throwing an OverflowException to parity what int.TryParse does.
The first case will probably result in an OutOfMemoryException but otherwise should be parsed as a valid set of integer digits.

Reply:
> The latter two should end up returning false for TryParse and throwing an OverflowException to parity what int.TryParse does.

Actually after thinking it over and talking with @bartonjs and @GrabYourPitchforks, I think that we should always throw `PlatformNotSupportedException` in this case as well. It is more consistent and allows us to expand in the future without an "additional" breaking change and avoids `false` being returned for a "technically valid" big integer that otherwise we simply don't support.